BREAKING.md

```diff
@@ -28,6 +29,60 @@ The `getQuorum()` method on `IContainer` and the `quorum` member of `IContainerC
 ### `IChannel.summarize` split into sync and async
 `IChannel` now has two summarization methods instead of a single synchronous `summarize`. `getAttachSummary` is synchronous to prevent channel modifications during summarization, `summarize` is asynchronous.
 
+### `wait()` methods deprecated on map and directory
+
+The `wait()` methods on `ISharedMap` and `IDirectory` have been deprecated and will be removed in an upcoming release.  To wait for a change to a key, you can replicate this functionality with a helper function that listens to the change events.
+
+```ts
+const directoryWait = async <T = any>(directory: IDirectory, key: string): Promise<T> => {
```

Comment:
I'm not sure I'd want to push people toward this. If it's required, it would be nice to understand scenario, as likely wait is the wrong way no matter what. Basically, this change may never happen (if we assume some other client makes such change). And it's much better to model it not as a wait, but as a (global) state machine that moves to a different state when a key is set (no promises).

Plus, solution likely needs to 
- handle failure points (container runtime being disposed)
- as any wait, it should have capability to report progress / be cancellable (we do not follow that today, but I think we should start doing it, and the sooner the better).
   - for example, if container is disconnected.

Reply:
I included a note below cautioning these risks -- I'm also not a fan of waiting (most of these usages are either unnecessary or what I'd call antipatterns).  But for the purpose of removing the built-in `.wait()` we need to provide a migration plan for current customers, I don't think it's a good idea to remove without guidance.

I don't expect that the removal of the built-in `.wait()` will hinder the development of any better alternatives, and in fact is a necessary stepping stone to get there.